### PR TITLE
Refactor: Mental exercise factory

### PR DESCRIPTION
**Pull request by AlexMi-Ha**

### CODE REVIEW

File: src/api/AMOGUS.Core/Factories/MentalExerciseFactory.cs [hunk 1]
```diff
@@ -104,74 +91,143 @@ public class MentalExerciseFactory : IExerciseFactory {
...
+            double weightedParams =
+                numOperandsMaxXp * numOperandsNormalized
                 + avgOperandsMaxXp * avgOperandsNormalized
                 + avgOperatorsMaxXp * avgOperatorsNormalized
-                + answerMaxXp * answerNormalized) / (double) Math.Max((amountMod5 - (expr.Operands.Length - amountMod5 + 1)), 1)) / (double) amountMod10), 0);
+                + answerMaxXp * answerNormalized;
 
-            expr.Difficulty = Math.Max(Math.Min((int) Math.Round(-Math.Exp((-4d / sumMaxXp) * (int) expr.Xp + 1.4) + 4), 4), 0);
+            // punish a lot mod 5s and mod 10s
+            double modWeight = 1d / Math.Max(amountMod5 - amountNotMod5 - 1, 1) * (amountMod10 + 1);
```
Comment (jas20202) [inline src/api/AMOGUS.Core/Factories/MentalExerciseFactory.cs:190]: Parenthesis missing: 1d / *(* mod5stuff * mod10stuff *)*;

**Review by jas20202 — CHANGES_REQUESTED** [review]
🧮

File: src/api/AMOGUS.Core/Factories/MentalExerciseFactory.cs [hunk 2]
```diff
@@ -104,74 +91,143 @@ public class MentalExerciseFactory : IExerciseFactory {
...
+         */
+        [ExcludeFromCodeCoverage]
+        private int GetRandomNumberOfOperands(bool insaneMode) {
+
+            return insaneMode ?
+                GenerateIntWithFallingProbability(3, 7)
+                : GenerateIntWithFallingProbability(2, 5);
+        }
+
+        /** <summary>
```
[the remark this answers is not in the document]
Comment (AlexMi-Ha) [reply src/api/AMOGUS.Core/Factories/MentalExerciseFactory.cs:130]: done

File: src/api/AMOGUS.Core/Factories/MentalExerciseFactory.cs (hunk 1 above)
Comment (AlexMi-Ha) [reply src/api/AMOGUS.Core/Factories/MentalExerciseFactory.cs]: done

**Review by Annalytic-programming — APPROVED** [review]
🥃

**Review by jas20202 — APPROVED** [review]
🦐

**Pull request merged**